core/src/main/java/org/apache/accumulo/core/client/mapreduce/lib/impl/InputConfigurator.java

```diff
@@ -517,6 +517,74 @@ public static Boolean isOfflineScan(Class<?> implementingClass, Configuration co
   }
 
   /**
+   * Controls the use of the {@link BatchScanner} in this job.
+   * Using this feature will group ranges by their source tablet per InputSplit and use BatchScanner to read them.
+   *
+   * <p>
+   * By default, this feature is <b>disabled</b>.
+   *
+   * @param implementingClass
+   *          the class whose name will be used as a prefix for the property configuration key
+   * @param conf
+   *          the Hadoop configuration object to configure
+   * @param enableFeature
+   *          the feature is enabled if true, disabled otherwise
+   * @since 1.7.0
+   */
+  public static void setBatchScan(Class<?> implementingClass, Configuration conf, boolean enableFeature) {
+    conf.setBoolean(enumToConfKey(implementingClass, Features.BATCH_SCANNER), enableFeature);
+  }
+
+  /**
+   * Determines whether a configuration has the BatchScanner feature enabled.
+   *
+   * @param implementingClass
+   *          the class whose name will be used as a prefix for the property configuration key
+   * @param conf
+   *          the Hadoop configuration object to configure
+   * @return true if the feature is enabled, false otherwise
+   * @since 1.7.0
+   * @see #setBatchScan(Class, Configuration, boolean)
+   */
+  public static Boolean isBatchScan(Class<?> implementingClass, Configuration conf) {
+    return conf.getBoolean(enumToConfKey(implementingClass, Features.BATCH_SCANNER), false);
+  }
+
+  /**
+   * Controls the use of the {@link BatchScanner} in this job.
+   * Using this feature will group ranges by their source tablet per InputSplit and use BatchScanner to read them.
+   *
+   * <p>
+   * By default, this feature is <b>disabled</b>.
+   *
+   * @param implementingClass
+   *          the class whose name will be used as a prefix for the property configuration key
+   * @param conf
+   *          the Hadoop configuration object to configure
+   * @param count
+   *          number of threads to use with BatchScanner
+   * @since 1.7.0
+   */
+  public static void setBatchScanThreads(Class<?> implementingClass, Configuration conf, int count) {
```

Comment:
>  Should it still remain for future refactors where BatchInputSplit gets initialized with ranges that span tablets?

Could remove the methods for now and a future changes could add them if needed.  Having them the way things work now is misleading.  

> Perhaps reduce default to one thread for now?

If the methods were removed, the implementation could do that.